slatedb/src/wal_buffer.rs

```diff
@@ -75,6 +76,9 @@ struct WalBufferManagerInner {
     /// Whenever a WAL is applied to Memtable and successfully flushed to remote storage,
     /// the immutable wal can be recycled in memory.
     last_applied_seq: Option<u64>,
+    /// Sequence numbers that should be tracked and inserted into the
+    /// manifest when the WAL/Memtable is flushed.
+    seq_tracker: TieredSequenceTracker,
```

Comment:
Feels odd to own this here (vs. an Arc or clone). Can we get rid of `seq_tracker(&self) -> TieredSequenceTracker` and have other modules that need `TieredSequenceTracker` clone from the builder and take the sequence tracker directly (e.g. `MemtableFlusher`)

Reply:
This is probably related to your other comment about the WAL buffer but I totally agree this is a little goofy. I'll try to find a better place for this that doesn't involve wiring in a whole new data path and also doesn't rely on storing it in in COW state (since it's modified frequently).